x/application/keeper/msg_server_undelegate_from_gateway_test.go

```diff
@@ -459,6 +460,42 @@ func TestMsgServer_UndelegateFromGateway_RedelegationAfterUndelegationAtTheSameS
 	require.Contains(t, gatewayAddressesAfterPruning, gatewayAddrToRedelegate)
 }
 
+func TestMsgServer_UndelegateFromGateway_UndelegateFromUnstakedGateway(t *testing.T) {
+	k, ctx := keepertest.ApplicationKeeper(t)
+	srv := keeper.NewMsgServerImpl(k)
+
+	undelegationHeight := int64(1)
+	sdkCtx, app, delegateAddr, pendingUndelegateFromAddr :=
+		createAppStakeDelegateAndUndelegate(ctx, t, srv, k, undelegationHeight)
+
+	require.Contains(t, app.DelegateeGatewayAddresses, delegateAddr)
+
+	// Assert that PendingUndelegations contains the pendingUndelegateFromAddr.
+	pendingUndelegateFromAddrs := make([]string, 0)
+	for _, pendingUndelegation := range app.PendingUndelegations {
+		pendingUndelegateFromAddrs = append(pendingUndelegateFromAddrs, pendingUndelegation.GatewayAddresses...)
+	}
+	require.Contains(t, pendingUndelegateFromAddrs, pendingUndelegateFromAddr)
+
+	// Increment the block height without moving to the next session.
+	sdkCtx = sdkCtx.WithBlockHeight(undelegationHeight + 1)
+
+	sdkCtx.EventManager().EmitTypedEvents(
+		&gwtypes.GatewayUnstaked{Address: delegateAddr},
+		&gwtypes.GatewayUnstaked{Address: pendingUndelegateFromAddr},
+	)
```

Comment:
#PUC that the functionality under tests depends on these events being emitted prior to calling the end blocker function.